chain/client/src/chunk_validation.rs

```diff
@@ -146,6 +147,62 @@ impl ChunkValidator {
         });
         Ok(())
     }
+
+    /// This function does two things
+    ///     1. Verifies if the chunk_endorsements have enough stake for chunk to be included in block.
+    ///     2. Collects the chunk_endorsement signatures from chunk_validators to include in block body.
+    /// We return an optional signature.
+    ///     None specifies we didn't have enough stake to include chunk in block.
+    ///     Some(signature) gives us the ordered chunk_endorsement signatures to include in block.
+    pub fn get_chunk_endorsement_signature(
+        &mut self,
+        chunk_header: &ShardChunkHeader,
+    ) -> Result<Option<ChunkEndorsementSignatures>, Error> {
+        // Get the chunk_endorsements for the chunk from our cache.
+        // Note that these chunk endorsements are already validated as part of process_chunk_endorsement.
+        // We can safely rely on the the following details
+        //    1. The chunk endorsements are from valid chunk_validator for this chunk.
+        //    2. The chunk endorsements signatures are valid.
+        let chunk_endorsements = self
+            .chunk_endorsements
+            .get_or_insert(chunk_header.chunk_hash(), || HashMap::new())
+            .unwrap();
```

Comment:
If we don't insert empty map here, then `self` doesn't have to be `mut` I think - this is generally expected from `get` method. And if result of `get` is None, we can early return as we currently do.

Reply:
I'd originally kept this as `mut` as I was updating the LRU cache and bringing the current chunk to the top. I can do a `peek` call without the `mut` as well, both work fine I guess.

Reply:
Removed `mut`